monitoring/rid_qualifier/aircraft_state_replayer.py

```diff
@@ -0,0 +1,145 @@
+
+            current_disk_rid_state_data['reference_time'] = test_reference_time.isoformat()
+            updated_timestamps_telemetry = []            
+
+            timestamp = test_start_offset.shift(seconds = 1)
+            
+            for telemetry_id, flight_telemetry in enumerate(current_disk_rid_state_data['flight_telemetry']['states']):
+
+                disk_telemetry_time_stamp_raw = flight_telemetry['timestamp']
+                disk_telemetry_time_stamp = arrow.get(disk_telemetry_time_stamp_raw)
+                reference_time_telemetry_time_stamp_delta_secs = (disk_telemetry_time_stamp - disk_reference_time).total_seconds()
+                
+                timestamp = timestamp.shift(seconds = reference_time_telemetry_time_stamp_delta_secs)
+                
+                flight_telemetry['timestamp'] = timestamp.isoformat()
+                updated_timestamps_telemetry.append(flight_telemetry)
+                        
+            flight_details =  current_disk_rid_state_data['flight_details']
+            operator_details = current_disk_rid_state_data['operator_details']
+            
+            operator_location = OperatorLocation(lat = operator_details['location']['lat'], lng = operator_details['location']['lng'])            
+            rid_flight_details = RIDFlightDetails(operator_id = str(uuid.uuid4()), operator_location = operator_location, operation_description = flight_details['operation_description'] , serial_number = flight_details['serial_number'], registration_number = flight_details['registration_number'])
+            test_flight_details = TestFlightDetails(effective_after = test_start_offset_isoformat,details = rid_flight_details)
+            test_flight = TestFlight(injection_id = str(uuid.uuid4()), telemetry = updated_timestamps_telemetry, details_responses = test_flight_details)
+            test_flight_deserialized = self.make_json_compatible(test_flight)
+            requested_flights.append(test_flight_deserialized)
+            test_payload = TestPayload(test_id = test_id, requested_flights = requested_flights)
+            test_payload_data_metadata = DeliverablePayloads(injection_path = test_injection_path, injection_payloads = test_payload)
+            all_test_payloads.append(test_payload_data_metadata)
+        
+        return all_test_payloads
+
+
+class TestHarness():
+    ''' A class to submit Aircraft RID State to the USS test endpoint '''
+
+    def __init__(self, auth_spec:str, injection_base_url:str):
+        
+        auth_adapter = make_auth_adapter(auth_spec)
+        self.uss_session = DSSTestSession(injection_base_url, auth_adapter)
+
+    def submit_test(self,uss_session:DSSTestSession, payload:DeliverablePayloads, test_injection_path:str) -> None:
+        
+        response = uss_session.put(url = test_injection_path, data=json.dumps(payload.injection_payloads), scope = ' '.join([rid.SCOPE_RID_QUALIFIER_INJECT ]))
```

Comment:
`data=<str>` [does not send a valid JSON payload](https://stackoverflow.com/questions/9733638/how-to-post-json-data-with-python-requests)

Reply:
Thanks it is now fixed and calls the `json` parameter